src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -516,6 +516,15 @@ ELSE pg_get_expr(indpred, cls.oid)
 
                         if (columnIndices.Any(i => i == 0))
                         {
+                            var expressions = record.GetValueOrDefault<string>("exprs");
+
+                            if (expressions?.StartsWith("to_tsvector") == true)
```

Comment:
This is somewhat "permissive", I'm a bit reluctant to identify any expression index starting with to_tsvector, as the range of actual expressions could be infinite and we risk scaffolding something that isn't a simple TsVector index (according to our definition) to that.

If you want to match the *exact* pattern that corresponds, I guess this could make sense (including the coalesce!), but I honestly don't think it's worth it. The general value in scaffolding this isn't that huge - I think it's better we just emitted a warning (which is already being done for expression indexes) and let the user do their thing.

Reply:
Am I right that you don't want this scaffolding? Without it tests fail. And I don't know how to disable this.

Perhaps changing clause to `expressions?.StartsWith("to_tsvector(")` can help. What do you think?

Reply:
Changed to `expressions?.StartsWith("to_tsvector(")`

Reply:
> Am I right that you don't want this scaffolding? Without it tests fail. And I don't know how to disable this.

Yeah, I'd rather we didn't do scaffold in "approximate" cases like this. Which tests are failing exactly? If we don't support scaffolding for this, there shouldn't be tests checking for that etc... I'll look into this after merging.